cypress/integration/tests/administrator/credentials/crud_sourceControl.test.ts

```diff
@@ -0,0 +1,50 @@
+import { hasToBeSkipped, login, preservecookies } from "../../../../utils/utils";
+import * as data from "../../../../utils/data_utils";
+import { CredentialsSourceControlUsername } from "../../../models/administrator/credentials/credentialsSourceControlUsername";
+import { CredentialsSourceControlKey } from "../../../models/administrator/credentials/credentialsSourceControlKey";
+
+describe("Validation of Source Control Credentials", () => {
+    const scCredsUsername = new CredentialsSourceControlUsername(data.getRandomWord(8));
+    scCredsUsername.description = data.getDescription();
+    scCredsUsername.username = data.getRandomWord(6);
+    scCredsUsername.password = data.getRandomWord(6);
+
+    const scCredsKey = new CredentialsSourceControlKey(data.getRandomWord(8));
+    scCredsKey.description = data.getDescription();
+    scCredsKey.key = data.getRandomWord(6);
+    scCredsKey.keyPassphrase = data.getRandomWord(6);
+
+    before("Login", function () {
+        // Prevent hook from running, if the tag is excluded from run
+        if (hasToBeSkipped("@tier2")) return;
+
+        // Perform login
+        login();
+    });
+
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+        preservecookies();
+    });
+
+    it("Creating source control credentials with username/password", () => {
+        scCredsUsername.create();
+    });
+
+    it("Editing source control credentials with username/password", () => {
+        scCredsUsername.edit();
```

Comment:
Let's assert if fields are edited correctly .

Reply:
The only way I can imaging to assert that changes were applied properly is to save changes, open for editing again and than compare if values were updated or not. Does it fit?

Reply:
@sshveta I have implemented validation that name is updated after editing and not updated if editing was canceled. Is that enough or do you want to validate that ALL fields are updated?